WebAPI/Controllers/StudentController.cs

```diff
@@ -83,6 +85,101 @@ public async Task<IActionResult> GetStudentTimetable(string userEmail)
             return Ok(timetable);
         }
 
+        [HttpGet("getStudentTimetableCalendar/{userEmail}")]
+        public async Task<IActionResult> GetStudentTimetableCalendar(string userEmail)
+        {
+            _logger.LogInformation($"[API getStudentTimetable] Getting timetable calendar for test student");
+            User user = await _userService.GetUserByEmailAsync(userEmail);
+            string studentId = user.Student.Id.ToString();
+            bool isValidGUID = Guid.TryParse(studentId, out Guid guid);
```

Comment:
I don't see the Student, what for type is `_user.Student.Id`, but it is strange you convert it to string and than to Guid

Reply:
I copied this from the method above, which retrieves the timetable. There is a duplicate now, but I am not sure whether to remove it or not.

Reply:
If _user.Student.Id is Guid, you don't have to convert it to string and back to Guid, that is all I want :-)

Reply:
and I am not sure about this, the TryParse gives me boolean and wants string

Reply:
Yes, returns if it is OK => boolean, input is string representation of GUID and output - GUID form is stored into guid parameter (is out)...
If you want we can discuss it by teams with live demo of code, or just put inside the breakpoint, check ` user.Student.Id` before and 'guid' after, if they are same you can replace both lines with
```
guid = user.Student.Id
```